src/pit/viper/_io.py

```diff
@@ -0,0 +1,91 @@
+"""Internal IO functions for viper."""
+import json
+import tomllib
+from pathlib import Path
+from typing import Any
+
+import yaml
+
+
+def read_config(path: Path) -> dict[str, Any]:
+    """Read a config file.
+
+    This function supports configs from JSON, TOML, and YAML files.
+
+    Parameters
+    ----------
+    path : Path
+       Path to the config file.
+
+    Returns
+    -------
+    dict[str, Any]
+        Config
+
+    Raises
+    ------
+    ValueError
+        Raised if the config file extension is not supported.
+    """
+    file_extension = path.suffix if path.suffix else path.name
+
+    if file_extension == ".json":
```

Comment:
Maybe a good use case for pattern matching.

Reply:
Nice suggestion. Looks cleaner.